### PR TITLE
Check if WithSpriteBody is available in GrantUpgradePower

### LINKED ISSUE #10005: Crash on "Bomber John" when trying to iron curtain minelayer. (current release)
```
Red Alert Mod at Version release-20150919
on map 8bc40b22e045f5561208cdb65cd182dc607efdc6 (Bomber John by Holloweye).
Operating System: Windows (Microsoft Windows NT 6.1.7601 Service Pack 1)
Runtime Version: .NET CLR 4.0.30319.34209
Exception of type `System.InvalidOperationException`: Actor does not have trait of type `OpenRA.Mods.Common.Traits.RenderBuilding`
   at OpenRA.TraitDictionary.TraitContainer`1.Get(UInt32 actor)
   at OpenRA.TraitDictionary.Get[T](Actor actor)
   at OpenRA.Actor.Trait[T]()
   at OpenRA.Mods.Common.Traits.GrantUpgradePower.Activate(Actor self, Order order, SupportPowerManager manager)
   at OpenRA.Mods.Common.Traits.SupportPowerInstance.Activate(Order order)
   at OpenRA.Mods.Common.Traits.SupportPowerManager.ResolveOrder(Actor self, Order order)
   at OpenRA.Network.UnitOrders.ProcessOrder(OrderManager orderManager, World world, Int32 clientId, Order order)
   at OpenRA.Network.OrderManager.Tick()
   at OpenRA.Game.InnerLogicTick(OrderManager orderManager)
   at OpenRA.Game.LogicTick()
   at OpenRA.Game.Loop()
   at OpenRA.Game.Run()
   at OpenRA.Program.Run(String[] args)
   at OpenRA.Program.Main(String[] args)
```

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/28263123-crash-on-bomber-john-when-trying-to-iron-curtain-minelayer-current-release?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by penev92**
Closes #10005.

### CODE REVIEW

**Comment by abcdefg30** [comment]
Looks reasonable. :+1:

**Comment by GraionDilach** [comment]
:+1:

File: OpenRA.Mods.Common/Traits/SupportPowers/GrantUpgradePower.cs [hunk 1]
```diff
@@ -56,7 +57,9 @@ public override void Activate(Actor self, Order order, SupportPowerManager manag
 		{
 			base.Activate(self, order, manager);
 
-			self.Trait<WithSpriteBody>().PlayCustomAnimation(self, info.GrantUpgradeSequence);
+			var wsb = self.TraitOrDefault<WithSpriteBody>();
+			if (wsb != null)
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/SupportPowers/GrantUpgradePower.cs]: This probably also wants a `HasSequence` check to match #10013.

**Comment by penev92** [comment]
Updated with a `HasSequence` check.

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/9d294e215671d8687b6fadcb970eb39f03898fc7...b01b2049343f0e8cd2d4482f5730a0d2540f285c)

**Pull request merged**